linkerd/stack/src/watch.rs

```diff
@@ -30,10 +11,23 @@ pub trait UpdateWatch<T>: Send + 'static {
 /// which define the logic for how to update the inner [`Service`].
 #[derive(Clone, Debug)]
 pub struct NewSpawnWatch<P, N> {
-    new_update: N,
+    inner: N,
     _marker: std::marker::PhantomData<fn(P)>,
 }
 
+#[derive(Clone, Debug)]
+pub struct NewWatchInto<P, N> {
+    inner: N,
+    _marker: std::marker::PhantomData<fn() -> P>,
+}
+
+#[derive(Clone, Debug)]
+pub struct NewInto<T, P, N> {
```

Comment:
naming quibble, take it or leave it: this looks like it specifically handles building a `P` from a `(T, U)` tuple...maybe `NewFromTuple` or `NewFromPair` (or something) would be a clearer name?

Reply:
also, very minor nit, but would it make sense for this to live in `stack` as its own middleware, since it's not really specific to watch? we might not need to use it anywhere else, so it's not a huge deal either way...

Reply:
it's already in stack, but for now this feels specific enough to commingle.